### ISSUE TITLE
[BUG] Storage over provisioning setting can be misinterpreted

### ISSUE THREAD

**Issue by tyzbit** [longhorn/longhorn#2425]
**Describe the bug**
I had a problem which required me to recreate my Kubernetes cluster.  I reused the masters and worker nodes without wiping the entire disks, but cleared out `/var/lib/rancher/longhorn` and the storage directory, `/mnt/external/longhorn/`.  When I create any volumes, they have scheduling failures.

I have a lot of the same symptoms as #2147

**To Reproduce**
1. Create a Kubernetes cluster and install Longhorn via Helm
2. Recreate the cluster using the same nodes
3. Reinstall Longhorn
4. Create a test volume

**Expected behavior**
Longhorn schedules volumes on reused nodes

**Log**
<details><summary>(Click) Manager log 1</summary>
<pre>
time="2021-03-30T21:42:44Z" level=info msg="Start overwriting built-in settings with customized values"
W0330 21:42:44.072047       1 client_config.go:541] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
time="2021-03-30T21:42:44Z" level=info msg="cannot list the content of the src directory /var/lib/rancher/longhorn/engine-binaries for the copy, will do nothing: Failed to execute: nsenter [--mount=/host/proc/1/ns/mnt --net=/host/proc/1/ns/net bash -c ls /var/lib/rancher/longhorn/engine-binaries/*], output , stderr, ls: cannot access '/var/lib/rancher/longhorn/engine-binaries/*': No such file or directory\n, error exit status 2"
I0330 21:42:44.087301       1 leaderelection.go:241] attempting to acquire leader lease  longhorn-system/longhorn-manager-upgrade-lock...
I0330 21:42:44.102639       1 leaderelection.go:251] successfully acquired lease longhorn-system/longhorn-manager-upgrade-lock
time="2021-03-30T21:42:44Z" level=info msg="Start upgrading"
time="2021-03-30T21:42:44Z" level=info msg="No API version upgrade is needed"
time="2021-03-30T21:42:44Z" level=info msg="Finish upgrading"
time="2021-03-30T21:42:44Z" level=info msg="Upgrade leader lost: showman"
W0330 21:42:44.180788       1 client_config.go:541] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I0330 21:42:44.190744       1 shared_informer.go:197] Waiting for caches to sync for longhorn datastore
E0330 21:42:44.262146       1 kubernetes_node_controller.go:232] couldn't get kubernetes node showman: node "showman" not found
I0330 21:42:44.391126       1 shared_informer.go:204] Caches are synced for longhorn datastore
W0330 21:42:44.391257       1 client_config.go:541] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
time="2021-03-30T21:42:44Z" level=info msg="Start Longhorn websocket controller" controller=longhorn-websocket node=showman
I0330 21:42:44.391290       1 shared_informer.go:197] Waiting for caches to sync for longhorn websocket
I0330 21:42:44.391300       1 shared_informer.go:204] Caches are synced for longhorn websocket
time="2021-03-30T21:42:44Z" level=info msg="Start Longhorn replica controller"
I0330 21:42:44.391329       1 shared_informer.go:197] Waiting for caches to sync for longhorn replicas
I0330 21:42:44.391336       1 shared_informer.go:204] Caches are synced for longhorn replicas
time="2021-03-30T21:42:44Z" level=info msg="Start kubernetes controller"
I0330 21:42:44.391413       1 shared_informer.go:197] Waiting for caches to sync for kubernetes
I0330 21:42:44.391426       1 shared_informer.go:204] Caches are synced for kubernetes
time="2021-03-30T21:42:44Z" level=info msg="Starting Longhorn instance manager controller"
I0330 21:42:44.391471       1 shared_informer.go:197] Waiting for caches to sync for longhorn instance manager
I0330 21:42:44.391480       1 shared_informer.go:204] Caches are synced for longhorn instance manager
time="2021-03-30T21:42:44Z" level=info msg="Start Longhorn share manager controller" controller=longhorn-share-manager node=showman
I0330 21:42:44.391524       1 shared_informer.go:197] Waiting for caches to sync for longhorn-share-manager-controller
I0330 21:42:44.391533       1 shared_informer.go:204] Caches are synced for longhorn-share-manager-controller
time="2021-03-30T21:42:44Z" level=info msg="Start Longhorn Engine Image controller"
I0330 21:42:44.391623       1 shared_informer.go:197] Waiting for caches to sync for longhorn engine images
I0330 21:42:44.391633       1 shared_informer.go:204] Caches are synced for longhorn engine images
time="2021-03-30T21:42:44Z" level=info msg="Start longhorn-kubernetes-pod-controller"
I0330 21:42:44.391771       1 shared_informer.go:197] Waiting for caches to sync for longhorn-kubernetes-pod-controller
I0330 21:42:44.391783       1 shared_informer.go:204] Caches are synced for longhorn-kubernetes-pod-controller
time="2021-03-30T21:42:44Z" level=info msg="Start Longhorn Kubernetes node controller"
I0330 21:42:44.392036       1 shared_informer.go:197] Waiting for caches to sync for longhorn kubernetes node
I0330 21:42:44.392050       1 shared_informer.go:204] Caches are synced for longhorn kubernetes node
time="2021-03-30T21:42:44Z" level=info msg=Start controller=longhorn-kubernetes-configmap-controller node=showman
I0330 21:42:44.392184       1 shared_informer.go:197] Waiting for caches to sync for longhorn-kubernetes-configmap-controller
I0330 21:42:44.392194       1 shared_informer.go:204] Caches are synced for longhorn-kubernetes-configmap-controller
time="2021-03-30T21:42:44Z" level=info msg="Start Longhorn node controller"
I0330 21:42:44.392298       1 shared_informer.go:197] Waiting for caches to sync for longhorn node
I0330 21:42:44.392306       1 shared_informer.go:204] Caches are synced for longhorn node
time="2021-03-30T21:42:44Z" level=info msg="Start Longhorn Setting controller" controller=longhorn-setting node=showman
I0330 21:42:44.392356       1 shared_informer.go:197] Waiting for caches to sync for longhorn settings
I0330 21:42:44.392363       1 shared_informer.go:204] Caches are synced for longhorn settings
time="2021-03-30T21:42:44Z" level=info msg="Start Longhorn engine controller" controller=longhorn-engine node=showman
I0330 21:42:44.392494       1 shared_informer.go:197] Waiting for caches to sync for longhorn engines
I0330 21:42:44.392500       1 shared_informer.go:204] Caches are synced for longhorn engines
time="2021-03-30T21:42:44Z" level=info msg="Start Longhorn volume controller" controller=longhorn-volume node=showman
I0330 21:42:44.392557       1 shared_informer.go:197] Waiting for caches to sync for longhorn engines
I0330 21:42:44.392563       1 shared_informer.go:204] Caches are synced for longhorn engines
time="2021-03-30T21:42:44Z" level=fatal msg="Error starting manager: settings.longhorn.io \"mkfs-ext4-parameters\" already exists"
</pre>
</details>

<details><summary>(Click) Manager log after restarting</summary>
<pre>
time="2021-03-30T21:42:57Z" level=info msg="Start overwriting built-in settings with customized values"
W0330 21:42:57.681900       1 client_config.go:541] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
time="2021-03-30T21:42:57Z" level=info msg="cannot list the content of the src directory /var/lib/rancher/longhorn/engine-binaries for the copy, will do nothing: Failed to execute: nsenter [--mount=/host/proc/1/ns/mnt --net=/host/proc/1/ns/net bash -c ls /var/lib/rancher/longhorn/engine-binaries/*], output , stderr, ls: cannot access '/var/lib/rancher/longhorn/engine-binaries/*': No such file or directory\n, error exit status 2"
I0330 21:42:57.694720       1 leaderelection.go:241] attempting to acquire leader lease  longhorn-system/longhorn-manager-upgrade-lock...
I0330 21:42:57.719489       1 leaderelection.go:251] successfully acquired lease longhorn-system/longhorn-manager-upgrade-lock
time="2021-03-30T21:42:57Z" level=info msg="Start upgrading"
time="2021-03-30T21:42:57Z" level=info msg="No API version upgrade is needed"
time="2021-03-30T21:42:57Z" level=info msg="Finish upgrading"
time="2021-03-30T21:42:57Z" level=info msg="Upgrade leader lost: showman"
W0330 21:42:57.861279       1 client_config.go:541] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I0330 21:42:57.864619       1 shared_informer.go:197] Waiting for caches to sync for longhorn datastore
I0330 21:42:57.964886       1 shared_informer.go:204] Caches are synced for longhorn datastore
W0330 21:42:57.965018       1 client_config.go:541] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
time="2021-03-30T21:42:57Z" level=info msg="Start Longhorn node controller"
I0330 21:42:57.965035       1 shared_informer.go:197] Waiting for caches to sync for longhorn node
I0330 21:42:57.965043       1 shared_informer.go:204] Caches are synced for longhorn node
time="2021-03-30T21:42:57Z" level=info msg="Start Longhorn engine controller" controller=longhorn-engine node=showman
I0330 21:42:57.965124       1 shared_informer.go:197] Waiting for caches to sync for longhorn engines
I0330 21:42:57.965132       1 shared_informer.go:204] Caches are synced for longhorn engines
time="2021-03-30T21:42:57Z" level=info msg="Start Longhorn replica controller"
I0330 21:42:57.965257       1 shared_informer.go:197] Waiting for caches to sync for longhorn replicas
I0330 21:42:57.965264       1 shared_informer.go:204] Caches are synced for longhorn replicas
time="2021-03-30T21:42:57Z" level=info msg="Starting Longhorn instance manager controller"
I0330 21:42:57.965358       1 shared_informer.go:197] Waiting for caches to sync for longhorn instance manager
I0330 21:42:57.965366       1 shared_informer.go:204] Caches are synced for longhorn instance manager
time="2021-03-30T21:42:57Z" level=info msg="Start Longhorn share manager controller" controller=longhorn-share-manager node=showman
I0330 21:42:57.965589       1 shared_informer.go:197] Waiting for caches to sync for longhorn-share-manager-controller
I0330 21:42:57.965666       1 shared_informer.go:204] Caches are synced for longhorn-share-manager-controller
time="2021-03-30T21:42:57Z" level=info msg="Start longhorn-kubernetes-pod-controller"
I0330 21:42:57.965862       1 shared_informer.go:197] Waiting for caches to sync for longhorn-kubernetes-pod-controller
I0330 21:42:57.965894       1 shared_informer.go:204] Caches are synced for longhorn-kubernetes-pod-controller
time="2021-03-30T21:42:57Z" level=info msg="Start Longhorn volume controller" controller=longhorn-volume node=showman
I0330 21:42:57.966032       1 shared_informer.go:197] Waiting for caches to sync for longhorn engines
I0330 21:42:57.966042       1 shared_informer.go:204] Caches are synced for longhorn engines
time="2021-03-30T21:42:57Z" level=info msg=Start controller=longhorn-kubernetes-configmap-controller node=showman
I0330 21:42:57.966129       1 shared_informer.go:197] Waiting for caches to sync for longhorn-kubernetes-configmap-controller
I0330 21:42:57.966136       1 shared_informer.go:204] Caches are synced for longhorn-kubernetes-configmap-controller
time="2021-03-30T21:42:57Z" level=info msg="Start Longhorn Setting controller" controller=longhorn-setting node=showman
I0330 21:42:57.966312       1 shared_informer.go:197] Waiting for caches to sync for longhorn settings
I0330 21:42:57.966322       1 shared_informer.go:204] Caches are synced for longhorn settings
time="2021-03-30T21:42:57Z" level=info msg="Start Longhorn Engine Image controller"
I0330 21:42:57.966444       1 shared_informer.go:197] Waiting for caches to sync for longhorn engine images
I0330 21:42:57.966453       1 shared_informer.go:204] Caches are synced for longhorn engine images
time="2021-03-30T21:42:57Z" level=info msg="Start Longhorn Kubernetes node controller"
I0330 21:42:57.966546       1 shared_informer.go:197] Waiting for caches to sync for longhorn kubernetes node
I0330 21:42:57.966555       1 shared_informer.go:204] Caches are synced for longhorn kubernetes node
time="2021-03-30T21:42:57Z" level=info msg="Start kubernetes controller"
I0330 21:42:57.966617       1 shared_informer.go:197] Waiting for caches to sync for kubernetes
I0330 21:42:57.966625       1 shared_informer.go:204] Caches are synced for kubernetes
time="2021-03-30T21:42:57Z" level=info msg="Start Longhorn websocket controller" controller=longhorn-websocket node=showman
I0330 21:42:57.966703       1 shared_informer.go:197] Waiting for caches to sync for longhorn websocket
I0330 21:42:57.966711       1 shared_informer.go:204] Caches are synced for longhorn websocket
time="2021-03-30T21:42:57Z" level=debug msg="Engine image longhornio/longhorn-engine:v1.1.0 is ready"
time="2021-03-30T21:42:57Z" level=info msg="Listening on 192.168.187.209:9500"
time="2021-03-30T21:42:57Z" level=info msg="Event(v1.ObjectReference{Kind:\"Node\", Namespace:\"longhorn-system\", Name:\"showman\", UID:\"55629897-cea5-4f5b-b998-5114e787bb11\", APIVersion:\"longhorn.io/v1beta1\", ResourceVersion:\"215721\", FieldPath:\"\"}): type: 'Warning' reason: 'Ready' Node showman is down: the manager pod longhorn-manager-5pz4q is not running"
time="2021-03-30T21:42:57Z" level=info msg="Event(v1.ObjectReference{Kind:\"Node\", Namespace:\"longhorn-system\", Name:\"showman\", UID:\"55629897-cea5-4f5b-b998-5114e787bb11\", APIVersion:\"longhorn.io/v1beta1\", ResourceVersion:\"215721\", FieldPath:\"\"}): type: 'Normal' reason: 'Schedulable' "
time="2021-03-30T21:42:57Z" level=debug msg="Prepare to create default instance manager instance-manager-e-7362d2ef, node: showman, default instance manager image: longhornio/longhorn-instance-manager:v1_20201216, type: engine"
time="2021-03-30T21:42:58Z" level=debug msg="Start backup store monitoring for s3://takeout-backups@us-east-1/"
time="2021-03-30T21:42:58Z" level=debug msg="Requeue longhorn-system/showman due to conflict: Operation cannot be fulfilled on nodes.longhorn.io \"showman\": the object has been modified; please apply your changes to the latest version and try again"
time="2021-03-30T21:42:58Z" level=info msg="Event(v1.ObjectReference{Kind:\"Node\", Namespace:\"longhorn-system\", Name:\"cameraman\", UID:\"ead0ca45-e24c-40cf-9f6b-2e15aef5ccad\", APIVersion:\"longhorn.io/v1beta1\", ResourceVersion:\"215492\", FieldPath:\"\"}): type: 'Normal' reason: 'Ready' Node cameraman is ready"
time="2021-03-30T21:42:58Z" level=debug msg="Instance Manager Controller showman picked up instance-manager-e-7362d2ef"
time="2021-03-30T21:42:58Z" level=warning msg="Starts to clean up then recreates pod for instance manager instance-manager-e-7362d2ef with state stopped"
time="2021-03-30T21:42:59Z" level=debug msg="Requeue longhorn-system/cameraman due to conflict: Operation cannot be fulfilled on nodes.longhorn.io \"cameraman\": the object has been modified; please apply your changes to the latest version and try again"
time="2021-03-30T21:42:59Z" level=info msg="Created instance manager pod instance-manager-e-7362d2ef for instance manager instance-manager-e-7362d2ef"
time="2021-03-30T21:43:00Z" level=info msg="Event(v1.ObjectReference{Kind:\"Node\", Namespace:\"longhorn-system\", Name:\"trainman\", UID:\"41ed93a9-050f-42d3-85e8-793d6dab40b6\", APIVersion:\"longhorn.io/v1beta1\", ResourceVersion:\"215455\", FieldPath:\"\"}): type: 'Normal' reason: 'Ready' Node trainman is ready"
time="2021-03-30T21:43:00Z" level=debug msg="Requeue longhorn-system/trainman due to conflict: Operation cannot be fulfilled on nodes.longhorn.io \"trainman\": the object has been modified; please apply your changes to the latest version and try again"
time="2021-03-30T21:43:01Z" level=debug msg="Start monitoring instance manager instance-manager-e-7362d2ef"
time="2021-03-30T21:43:03Z" level=info msg="Event(v1.ObjectReference{Kind:\"Node\", Namespace:\"longhorn-system\", Name:\"anchorman\", UID:\"bc1338cc-a334-4fa2-a9fd-6e97cd427c00\", APIVersion:\"longhorn.io/v1beta1\", ResourceVersion:\"215540\", FieldPath:\"\"}): type: 'Normal' reason: 'Ready' Node anchorman is ready"
time="2021-03-30T21:43:03Z" level=debug msg="Requeue longhorn-system/anchorman due to conflict: Operation cannot be fulfilled on nodes.longhorn.io \"anchorman\": the object has been modified; please apply your changes to the latest version and try again"
time="2021-03-30T21:43:06Z" level=info msg="Event(v1.ObjectReference{Kind:\"Node\", Namespace:\"longhorn-system\", Name:\"showman\", UID:\"55629897-cea5-4f5b-b998-5114e787bb11\", APIVersion:\"longhorn.io/v1beta1\", ResourceVersion:\"215764\", FieldPath:\"\"}): type: 'Normal' reason: 'Ready' Node showman is ready"
time="2021-03-30T21:43:06Z" level=debug msg="Requeue longhorn-system/showman due to conflict: Operation cannot be fulfilled on nodes.longhorn.io \"showman\": the object has been modified; please apply your changes to the latest version and try again"
</pre>
</details>

**Environment:**
 - Longhorn version: v1.1.0
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: kubeadm, v1.19.7
   - Number of management node in the cluster: 3
   - Number of worker node in the cluster: 6 total, 3 with replicas
 - Node config
   - OS type and version: Ubuntu 20.04.1 LTS
   - CPU per node: 8
   - Memory per node: 16GB
   - Disk type(e.g. SSD/NVMe): NVMe (Samsung 970 Evo Plus)
   - Network bandwidth between the nodes: 2.5Gbps
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): Baremetal
 - Number of Longhorn volumes in the cluster: 1

**Comment by tyzbit** [comment]
[support bundle](https://github.com/longhorn/longhorn/files/6233283/longhorn-support-bundle_5be709ef-a54d-46ea-971f-a9f7c4fb4b26_2021-03-31T01-47-43Z.zip)

**Comment by innobead** [comment]
Thanks for the reporting. 

The settings CR seems not to be removed. Please make sure to follow the [uninstall process](https://longhorn.io/docs/1.1.0/deploy/uninstall/) to clean up the old cluster nodes.

The workaround, please delete the `setting.longhorn.io` resources by yourself, then restart the longhorn-manager daemon to let the manager recreate the setting. 

cc @cclhsu please follow up.

**Comment by tyzbit** [comment]
When I uninstalled, I did `helm uninstall`.  After uninstalling again, I double checked that the settings custom resources were gone:
```
~|main ⇒ (⎈ |takeout:longhorn-system) kubectl get settings.longhorn.io
error: the server doesn't have a resource type "settings"
```

After reinstalling with helm, the issue persists.  I even ran the CRD removing script on that page.

**Comment by tyzbit** [comment]
I figured it out (spoiler: it was self-inflicted).  I was setting `storageOverProvisioningPercentage` to `0` instead of `100`.  At the time I thought the value was what percentage **over 100%** to add.  In my previous installation I must have modified this value correctly, but this time I pre-set the values via the helm chart in a Flux HelmRelease, which did not have the helpful tooltip `The over-provisioning percentage defines how much storage can be allocated relative to the hard drive's capacity` to remind me that it's actually the literal provisioning percentage, not the OVER provisioning percentage.

Leaving open in case there's discussion to be had about if other people could get confused by the wording of the value like I did, but otherwise feel free to close.
